ditto/readers/abstract_reader.py

```diff
@@ -476,6 +530,236 @@ def get_sequence_impedances(self, dist_matrix, GMR_list, r_list):
+                )
+                gmr_neutral = (
+                    concentric_neutral_gmr
+                    * concentric_neutral_nstrand
+                    * (radius ** (concentric_neutral_nstrand - 1))
+                ) ** (1 / concentric_neutral_nstrand)
+                r_neutral = self.convert_from_meters(
+                    concentric_neutral_resistance_meters, "mi", inverse=True
+                ) / float(concentric_neutral_nstrand)
+                gmrs.append(self.convert_from_meters(gmr_meters, "ft"))
+                rs.append(
+                    self.convert_from_meters(resistance_meters, "ft", inverse=True)
+                )
+                gmrs_neutral.append(gmr_neutral)
+                rs_neutral.append(r_neutral)
+                radius_neutral.append(radius)
+
+            distance_matrix_cn = [
+                [None for i in range(2 * nphases)] for j in range(2 * nphases)
+            ]
+            for i in range(nphases):
+                for j in range(nphases):
+                    distance_matrix_cn[i][j] = distance_matrix_feet[i][j]
+                    distance_matrix_cn[i + nphases][j + nphases] = distance_matrix_feet[
+                        i
+                    ][j]
+
+                    # As per Example 4.2 of Kersting. phase-neutral equals the concentric-neutral radius for same cable
+                    if i == j:
+                        distance_matrix_cn[i + nphases][j] = radius_neutral[i]
+                        distance_matrix_cn[i][j + nphases] = radius_neutral[i]
+                    # As per Example 4.2 of Kersting. phase-neutral = phase-phase distance for different cables
+                    else:
+                        distance_matrix_cn[i + nphases][j] = distance_matrix_feet[i][j]
+                        distance_matrix_cn[i][j + nphases] = distance_matrix_feet[i][j]
+            gmrs_cn = []
+            rs_cn = []
+            for i in range(nphases):
+                gmrs_cn.append(gmrs[i])
+                rs_cn.append(rs[i])
+            for i in range(nphases):
+                gmrs_cn.append(gmrs_neutral[i])
+                rs_cn.append(rs_neutral[i])
+
+            impedance_matrix_imperial = self.get_primitive_impedance_matrix(
+                np.array(distance_matrix_cn), gmrs_cn, rs_cn
+            )
+            # Now Kron reduce out the nphases neutrals
+            neutrals = []
+            for i in range(nphases):
+                neutrals.append(i + nphases)
+            impedance_matrix_imperial = self.kron_reduction(
+                impedance_matrix_imperial, neutrals=neutrals
+            )
+
+            for i in range(len(impedance_matrix)):
+                for j in range(len(impedance_matrix)):
+                    impedance_matrix[i][j] = self.convert_to_meters(
+                        impedance_matrix_imperial[i][j], "mi", inverse=True
+                    )  # works for complex
+
+        return impedance_matrix
+
```

Comment:
This `get_phase_impedances` function is way too long. Can this be refactored into two or more blocks as different functions? Also, this is not being tested.